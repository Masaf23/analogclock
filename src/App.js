import React, { Component } from 'react';
import Form from './Form';
import AnalogClock from './AnalogClock.js'


class App extends Component {

  interval = null;
  constructor(props) {
    super(props);
    this.state = {
      options: {
        useCustomTime: false,
        width: "300px",
        border: true,
        borderColor: "#2e2e2e",
        baseColor: "#17a2b8",
        centerColor: "#459cff",
        centerBorderColor: "#ffffff",
        handColors: {
          second: "#d81c7a",
          minute: "#ffffff",
          hour: "#ffffff"
        },
      }
    };
  }


  updateClock() {
    let ausTime = new Date().toLocaleString("en-US", { timeZone: "Australia/Brisbane" });
    let date = new Date(ausTime);

    this.setState({
      'options': {
        ...this.state.options,
        seconds: date.getSeconds(),
        minutes: date.getMinutes(),
        hours: date.getHours()
      }
    })
  }

  customizeClock(options) {
    let _options = options;
    if (_options.useCustomTime) {
      this.interval = setInterval(() => this.updateClock(), 1000);
    } else {
      clearInterval(this.interval);
      delete _options.seconds;
      delete _options.minutes;
      delete _options.hours;
    }
    this.setState({ options: { ..._options } });
  }

  render() {
    return (
      <div>
        <div className="row">
          <Form defaultOptions={this.state.options} customizeClock={(options) => this.customizeClock(options)} />
        </div>
        <div className="row mt-3">
          <div className="col-6 border-right">
            <h4><i>Options</i></h4>
            <textarea className="form-control" id="selected-options" rows="12" readOnly value={JSON.stringify(this.state.options, null, 2)}></textarea>
          </div>
          <div className="col-6">
            <h4><i>Preview</i></h4>
            <AnalogClock {...this.state.options} />
            {this.state.options.useCustomTime ? <p style={{ paddingTop: '5px', paddingLeft: '20px' }}><i>Timezone: Australia/Brisbane</i></p> : null}
          </div>
        </div>
      </div>
    );
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }
}

export default App;