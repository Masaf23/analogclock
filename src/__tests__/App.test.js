import React from "react";
import App from '../App'
import { shallow } from "enzyme";

describe('App component', () => {

    it('matches snapshot', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot()
    });

    it('matches snapshot when options are changed', () => {
        const wrapper = shallow(<App />);
        let options = {
            useCustomTime: true,
            seconds: 10,
            minutes: 10,
            hours: 10,
            baseColor: "red",
            border: false,
            borderColor: "#fff",
            centerBorderColor: "black",
            centerColor: "blue"
        }
        wrapper.setState({ 'options': { ...wrapper.state().options, ...options } });
        expect(wrapper).toMatchSnapshot()
    });
})