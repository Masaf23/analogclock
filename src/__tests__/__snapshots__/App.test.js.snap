// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <Form
      customizeClock={[Function]}
      defaultOptions={
        Object {
          "baseColor": "#17a2b8",
          "border": true,
          "borderColor": "#2e2e2e",
          "centerBorderColor": "#ffffff",
          "centerColor": "#459cff",
          "handColors": Object {
            "hour": "#ffffff",
            "minute": "#ffffff",
            "second": "#d81c7a",
          },
          "useCustomTime": false,
          "width": "300px",
        }
      }
    />
  </div>
  <div
    className="row mt-3"
  >
    <div
      className="col-6 border-right"
    >
      <h4>
        <i>
          Options
        </i>
      </h4>
      <textarea
        className="form-control"
        id="selected-options"
        readOnly={true}
        rows="12"
        value="{
  \\"useCustomTime\\": false,
  \\"width\\": \\"300px\\",
  \\"border\\": true,
  \\"borderColor\\": \\"#2e2e2e\\",
  \\"baseColor\\": \\"#17a2b8\\",
  \\"centerColor\\": \\"#459cff\\",
  \\"centerBorderColor\\": \\"#ffffff\\",
  \\"handColors\\": {
    \\"second\\": \\"#d81c7a\\",
    \\"minute\\": \\"#ffffff\\",
    \\"hour\\": \\"#ffffff\\"
  }
}"
      />
    </div>
    <div
      className="col-6"
    >
      <h4>
        <i>
          Preview
        </i>
      </h4>
      <AnalogClock
        baseColor="#17a2b8"
        border={true}
        borderColor="#2e2e2e"
        centerBorderColor="#ffffff"
        centerColor="#459cff"
        handColors={
          Object {
            "hour": "#ffffff",
            "minute": "#ffffff",
            "second": "#d81c7a",
          }
        }
        useCustomTime={false}
        width="300px"
      />
    </div>
  </div>
</div>
`;

exports[`App component matches snapshot when options are changed 1`] = `
<div>
  <div
    className="row"
  >
    <Form
      customizeClock={[Function]}
      defaultOptions={
        Object {
          "baseColor": "red",
          "border": false,
          "borderColor": "#fff",
          "centerBorderColor": "black",
          "centerColor": "blue",
          "handColors": Object {
            "hour": "#ffffff",
            "minute": "#ffffff",
            "second": "#d81c7a",
          },
          "hours": 10,
          "minutes": 10,
          "seconds": 10,
          "useCustomTime": true,
          "width": "300px",
        }
      }
    />
  </div>
  <div
    className="row mt-3"
  >
    <div
      className="col-6 border-right"
    >
      <h4>
        <i>
          Options
        </i>
      </h4>
      <textarea
        className="form-control"
        id="selected-options"
        readOnly={true}
        rows="12"
        value="{
  \\"useCustomTime\\": true,
  \\"width\\": \\"300px\\",
  \\"border\\": false,
  \\"borderColor\\": \\"#fff\\",
  \\"baseColor\\": \\"red\\",
  \\"centerColor\\": \\"blue\\",
  \\"centerBorderColor\\": \\"black\\",
  \\"handColors\\": {
    \\"second\\": \\"#d81c7a\\",
    \\"minute\\": \\"#ffffff\\",
    \\"hour\\": \\"#ffffff\\"
  },
  \\"seconds\\": 10,
  \\"minutes\\": 10,
  \\"hours\\": 10
}"
      />
    </div>
    <div
      className="col-6"
    >
      <h4>
        <i>
          Preview
        </i>
      </h4>
      <AnalogClock
        baseColor="red"
        border={false}
        borderColor="#fff"
        centerBorderColor="black"
        centerColor="blue"
        handColors={
          Object {
            "hour": "#ffffff",
            "minute": "#ffffff",
            "second": "#d81c7a",
          }
        }
        hours={10}
        minutes={10}
        seconds={10}
        useCustomTime={true}
        width="300px"
      />
      <p
        style={
          Object {
            "paddingLeft": "20px",
            "paddingTop": "5px",
          }
        }
      >
        <i>
          Timezone: Australia/Brisbane
        </i>
      </p>
    </div>
  </div>
</div>
`;
